ospf6d/ospf6_interface.c

```diff
@@ -460,6 +461,9 @@ void ospf6_interface_connected_route_update(struct interface *ifp)
 		route->path.area_id = oi->area->area_id;
 		route->path.type = OSPF6_PATH_TYPE_INTRA;
 		route->path.cost = oi->cost;
+		if (oi->type == OSPF_IFTYPE_LOOPBACK)
```

Comment:
Why is this just loopback?
LA-bit
      The "local address" capability bit.  If set, the prefix is
      actually an IPv6 interface address of the Advertising Router.
      Advertisement of local interface addresses is described in
      Section 4.4.3.9.  An implementation MAY also set the LA-bit for
      prefixes advertised with a host PrefixLength (128).

Reply:
That's a good point.

Marking the PR as WIP until the comment is addressed.  Thanks.

Reply:
I really dislike the way "we" write RFCs ... the language here is about as clear as mud. 

According to 5340, A.4.1.1, the LA-bit is used to describe a `prefix [that] is actually an IPv6 interface address of the Advertising Router.` This might imply there is an overlapping longer prefix "covering" the same space, which means the interface address itself does not need to be advertised... but the reasoning isn't really spelled out. It does seem to be restricted to host prefixes, however, since this section also says the LA-bit MAY be set on any other prefix with a host length.

The practical effect seems to be that such routes will not be copied into a type 2 intra-area LSA when advertised by a router that is not the DR, according 4.4.3.9 --

`Prefixes having the NU-bit and/or LA-bit set in their Options field SHOULD NOT be copied, nor should link-local addresses be copied.`

But interfaces locally attached to the DR with the LA-bit should/can be advertised in the type2- intra-area LSA --

`However, any prefixes that would normally have the LA-bit set SHOULD be advertised independent of whether or not the interface is advertised as a transit link.`

This seems to be designed to prevent these host routes from being advertised as if they are connected to the pseudonode on multiaccess links -- they should only be advertised by the router to which they are connected.

In another section, 5340 states that if the router is the head or tail end of a virtual link, the endpoint address should be marked with the LA-bit and advertised with a metric of 0. I didn't go look up how this is used to discover virtual link endpoints.

5340 is updated by several RFCs, including 6860 and 7503, but neither of these mention the LA-bit. However... 8362 does modify the LA-bit ...

`The prefix options are extended from Appendix A.4.1.1 [OSPFV3].  The applicability of the LA-bit is expanded, and it SHOULD be set in Inter-Area-Prefix TLVs and MAY be set in External-Prefix TLVs when the advertised host IPv6 address, i.e., PrefixLength = 128 for the IPv6 Address Family or PrefixLength = 32 for the IPv4 Address Family [OSPFV3-AF], is an interface address.  In RFC 5340, the LA-bit is only set in Intra-Area-Prefix-LSAs (Section 4.4.3.9 of [OSPFV3]). This will allow a stable address to be advertised without having to configure a separate loopback address in every OSPFv3 area.`

And then it goes on to mention using the LA-bit in relation to anycast addresses, but I didn't chase that one down.

Bottom line -- The LA-bit should be set on locally connected interfaces with a host prefix-length so they don't end up in type-2 intra-area LSAs. Other than this, the LA-bit may be set on any host length prefix. I don't see any real reason NOT to set it for all host length prefixes, but I might still be missing some specific reason someplace (?).

Maybe this helps job someone's memory...